mmdet/datasets/pipelines/loading.py

```diff
@@ -607,26 +607,20 @@ def __call__(self, results):
             assert 'gt_masks' in results
             gt_masks = results['gt_masks']
             instance_num = len(gt_masks)
-
+        keep = np.ones(instance_num, dtype=bool)
```

Comment:
How about add a assert before to avoid some potential error?
such as:
```
instance_num = None
if self.by_box:
    if instance_num is not None:
        assert instance_num == box_num
if self.by_mask:
        if instance_num is not None:
            assert instance_num == mask_num

```